src/klein/_form.py

```diff
@@ -138,9 +144,13 @@ def asTags(self):
         @return: A new set of tags to include in a template.
         @rtype: iterable of L{twisted.web.template.Tag}
         """
+        value = self.value
+        if self.value is None:
+            # FIXME: mypy says this is unreachable because self.value is not
+            # optional.  Should it be?
+            value = ""  # type: ignore[misc]
```

Comment:
@glyph I think this was your doing.  Should `self.value` be optional in this class?

Reply:
It's sort of … conditionally optional?  If `required` is True it probably should be non-optional, but maybe it's better to just say that it is `Optional` for simplicity's sake.

Reply:
OK.  We could add runtime validation if we care, but I don't see that being solved using type hints.